test/wasm/wasm_api.test.js

```diff
@@ -0,0 +1,62 @@
+const { test, expect } = require('@playwright/test');
+
+test.describe('WASM Sketcher API', () => {
+
+    // Increase timeout for this test suite to 60 seconds
+    test.setTimeout(60000);
+
+    test.beforeEach(async ({ page }) => {
+        // Navigate to the page that loads the WASM module and
+        // wait for the WASM module to be fully loaded and available
+        await page.goto('/wasm_shell.html');
+        await page.waitForFunction('window.Module');
+
+        // Verify that the module is not null or undefined
+        const moduleLoaded = await page.evaluate(() => typeof window.Module !== 'undefined');
+        expect(moduleLoaded).toBe(true);
+    });
+
+    test('Roundtrip SMILES through the sketcher', async ({ page }) => {
+        const smiles_input = "c1ccccc1";
+        const exportedText = await page.evaluate(async (text) => {
+            Module.sketcher_import_text(text);
+            // Wait for the change to propagate
+            await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
I think that API is blocking so we shouldn't have to wait extra. Unless calling `sketcher_import_text` and `sketcher_export_text` one after the other in C++ land would also not work.